pkg/kubefedctl/enable/enable.go

```diff
@@ -358,8 +358,7 @@ func qualifiedAPIResourceName(resource metav1.APIResource) string {
 }
 
 func federatedTypeCRD(typeConfig typeconfig.Interface, accessor schemaAccessor, shortNames []string) *apiextv1b1.CustomResourceDefinition {
-	templateSchema := accessor.templateSchema()
-	schema := federatedTypeValidationSchema(templateSchema)
+	schema := federatedTypeValidationSchema(nil)
```

Comment:
It's not a good sign that this PR is passing CI, since `templateSchema` should still be provided for scalable resources to ensure that `spec.retainReplicas` is added to the validation schema. It's my fault, though, since I neglected to write a test for that case.

I think the easiest path forward is to still compute `templateSchema` and instead update  `federatedTypeValidationSchema` to avoid setting `Properties` for `spec.template` (i.e. not set the validation schema past the field being of type `object`). It probably makes sense to add a TODO to remind us to find a simpler way to detect that a resource is scalable than having to compute the entire schema.

Reply:
@marun, have worked on the comment. PTAL, Thanks